### PR TITLE
Use the Yaml scrubber automatically

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

The default scrubber can't handle sensitive data in Yaml list. So whenever we include a Yaml file into the flare or send the Agent configuration to inventories we now use the Yaml scrubber.

This scubs some uncommon configuration such as:

```
some_sensitive_data:
  - data1
  - data2
```
### Describe how to test/QA your changes

Set a list to be scrub using `flare_stripped_keys` and check the data is scrubbed in flares and in the Datadog UI.

Example:
```
histogram_aggregates:
  - max
  - median
  - avg
  - count

flare_stripped_keys:
  - histogram_aggregates
```

Should appears as:
```
histogram_aggregates: "*******"

flare_stripped_keys:
  - histogram_aggregates
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/fix-yaml-scrubbing-06bb43a799494065.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    The scubbing logic for configurations now scrub YAML list. This fixes ``flare_stripped_keys`` not working on YAML
```
Comment (hestonhoffman) [inline releasenotes/notes/fix-yaml-scrubbing-06bb43a799494065.yaml:4]:
```suggestion
    The scubbing logic for configurations now scrubs YAML lists. This fixes ``flare_stripped_keys`` not working on YAML.
```

**Review by hestonhoffman — APPROVED** [review]
Small edit

File: releasenotes/notes/fix-yaml-scrubbing-06bb43a799494065.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    The scubbing logic for configurations now scrubs YAML lists. This fixes ``flare_stripped_keys`` not working on YAML.
+    list.
```
Comment (pgimalac) [inline releasenotes/notes/fix-yaml-scrubbing-06bb43a799494065.yaml:5]:
```suggestion
```

File: pkg/util/scrubber/test/conf_multiline.yaml [hunk 3]
```diff
@@ -19,3 +19,9 @@ second_token: >
   123465749798
 
 third_token: 123465749798
+forth_token: True
```
Comment (pgimalac) [inline pkg/util/scrubber/test/conf_multiline.yaml:22]:
```suggestion
fourth_token: True
```

File: pkg/util/scrubber/test/conf_multiline_scrubbed.yaml [hunk 4]
```diff
@@ -4,6 +4,8 @@ api_key3: '***************************abbbb'
 app_key: '***********************************abbbb'
 app_key2: '***********************************abbbb'
 app_key3: '***********************************abbbb'
+forth_token: "********"
```
Comment (pgimalac) [inline pkg/util/scrubber/test/conf_multiline_scrubbed.yaml:7]:
```suggestion
fourth_token: "********"
```

**Review by pgimalac — APPROVED** [review]
Beware that there are some typing issues that appear in the CI

File: releasenotes/notes/fix-yaml-scrubbing-06bb43a799494065.yaml (hunk 2 above)
Comment (hush-hush) [reply releasenotes/notes/fix-yaml-scrubbing-06bb43a799494065.yaml:5]: We need the list, but I removed the dot on the previous line.

**Pull request merged**